utils/celery_worker.py

```diff
@@ -83,96 +85,118 @@ def setup_periodic_tasks(sender, **kwargs):
+            task_status='Failed',
             timestamp=scan_time,
         )
 
-    # Save the reports to storage
-    storage_handler.store(results, wait=False)
-    storage_handler.close()
-
-    return results
-
 
-@app.task(base=Batches, flush_every=api_config['batch_size'], flush_interval=api_config['batch_interval'])
-def multiscanner_celery(requests, *args, **kwargs):
+@app.task(base=MultiScannerTask)
+def multiscanner_celery(file_, original_filename, task_id, file_hash, metadata, config=multiscanner.CONFIG, module_list=None):
     '''
-    Queue up multiscanner tasks and then run a batch of them at a time for
-    better performance.
+    Queue up multiscanner tasks
 
     Usage:
     from celery_worker import multiscanner_celery
     multiscanner_celery.delay(full_path, original_filename, task_id, metdata,
-                              hashed_filename, config)
+                              hashed_filename, metadata, config)
     '''
     # Initialize the connection to the task DB
     db.init_db()
 
     files = {}
-    for request in requests:
-        file_ = request.args[0]
-        original_filename = request.args[1]
-        task_id = request.args[2]
-        file_hash = request.args[3]
-        metadata = request.args[4]
-        # print('\n\n{}{}Got file: {}.\nOriginal filename: {}.\n'.format('='*48, '\n', file_hash, original_filename))
-        files[file_] = {
-            'original_filename': original_filename,
-            'task_id': task_id,
-            'file_hash': file_hash,
-            'metadata': metadata,
-        }
-
-    celery_task(files)
+    logger.info('\n\n{}{}Got file: {}.\nOriginal filename: {}.\n'.format('='*48, '\n', file_hash, original_filename))
+    files[file_] = {
+        'original_filename': original_filename,
+        'task_id': task_id,
+        'file_hash': file_hash,
+        'metadata': metadata,
+    }
+
+    # Get the storage config
+    storage_conf = multiscanner.common.get_config_path(config, 'storage')
+    storage_handler = multiscanner.storage.StorageHandler(configfile=storage_conf)
 
+    resultlist = multiscanner.multiscan(
+        list(files),
+        configfile=config,
+        module_list=module_list
+    )
+    results = multiscanner.parse_reports(resultlist, python=True)
+
+    scan_time = datetime.now().isoformat()
+
+    # Get the Scan Config that the task was run with and
+    # add it to the task metadata
+    scan_config_object = configparser.SafeConfigParser()
+    scan_config_object.optionxform = str
+    scan_config_object.read(config)
+    full_conf = common.parse_config(scan_config_object)
+    sub_conf = {}
+    for key in full_conf:
+        if key == 'main':
+            continue
+        sub_conf[key] = {}
+        sub_conf[key]['ENABLED'] = full_conf[key]['ENABLED']
+
+    # Count number of modules enabled out of total possible
+    # and add it to the Scan Metadata
+    total_enabled = 0
+    total_modules = 0
+    for key in sub_conf:
```

Comment:
could you calculate total/enabled modules in the previous for loop instead of needing a second one? I guess this is slightly easier to read.